tests/test_bc.py

```diff
@@ -19,24 +20,49 @@ def venv():
     return venv
 
 
-@pytest.fixture(params=[False, True])
-def trainer(request, venv):
-    convert_dataset = request.param
+@pytest.fixture(params=[32, 50])
+def batch_size(request):
+    return request.param
+
+
+@pytest.fixture(params=[True, False])
+def use_ducktyped_dataloader(request):
+    return request.param
+
+
+class DucktypedDataset:
+    """Used to check that any iterator over Dict[str, Tensor] works with BC."""
+
+    def __init__(self, transitions, batch_size):
+        self.trans = transitions
+        self.batch_size = batch_size
+
+    def __iter__(self):
+        for start in range(0, len(self.trans), self.batch_size):
+            d = dict(obs=self.trans.obs, acts=self.trans.acts)
+            d = {k: th.from_numpy(v) for k, v in d.items()}
+            yield d
+
+
+@pytest.fixture
+def trainer(batch_size, venv, use_ducktyped_dataloader):
     rollouts = types.load(ROLLOUT_PATH)
-    data = rollout.flatten_trajectories(rollouts)
-    if convert_dataset:
-        data = datasets.TransitionsDictDatasetAdaptor(
-            data, datasets.EpochOrderDictDataset
-        )
-    return bc.BC(venv.observation_space, venv.action_space, expert_data=data)
+    trans = rollout.flatten_trajectories(rollouts)
+    if not use_ducktyped_dataloader:
```

Comment:
Nit: it feels more natural to me to write `if use_ducktyped_dataloader` and swap the two branches.